### PR TITLE
Plugin breaks on ruby 1.9.x

### PR DESCRIPTION
**Pull request by ppp0**

### CODE REVIEW

**Comment by ppp0** [comment]
We had problems in our CI for Wheezy-based servers. `fluentd` is compatible with rb v.1.9.x up to versions `< 0.14` see https://github.com/cargomedia/puppet-packages/pull/1347 and https://github.com/cargomedia/puppet-packages/pull/1342 (solves for fluentd)

@y-ken Excuse me, do you think it would be possible to release this as backwards-compatible plugin version? Thank you very much

**Comment by ppp0** [comment]
This would have to be released before #8

**Comment by ppp0** [comment]
This is because of https://github.com/fluent/fluentd/issues/888#issuecomment-223298220

**Comment by y-ken** [comment]
I can release for ruby-1.9 final compatible version but there are no update for it.
I'll keep maintain for fluentd v0.12 and v0.14.

**Comment by ppp0** [comment]
Great! That is a great help to us folks running fluentd on Debian Wheezy! Thank you very much

**Comment by y-ken** [comment]
Is it ok to release tomorrow? If you get hurry, I'm going to begin it soon.

**Comment by ppp0** [comment]
I am in your hands :) - you can choose

File: fluent-plugin-sentry.gemspec [hunk 1]
```diff
@@ -23,6 +23,6 @@ Gem::Specification.new do |spec|
     spec.add_development_dependency "test-unit", '~> 3'
   end
   
-  spec.add_runtime_dependency "fluentd"
+  spec.add_runtime_dependency "fluentd", "~> 0.12.0"
```
Comment (y-ken) [inline fluent-plugin-sentry.gemspec]: How about adding ruby version switch?

**Comment by y-ken** [comment]
It won't brake if you install fluentd.

**Comment by ppp0** [comment]
@y-ken I actually tested this several times.. ? 

I don't know how to reproduce your findings :( . Please have a look at what I did:

```sh
root@host:/home/admin$ ruby --version
ruby 1.9.3p194 (2012-04-20 revision 35410) [x86_64-linux]

root@host:/home/admin$ gem list fluentd*

*** LOCAL GEMS ***

fluent-mixin-config-placeholders (0.3.1)
fluent-plugin-forest (0.3.0)
fluent-plugin-loggly (0.0.7)
fluent-plugin-record-modifier (0.4.1)
fluentd (0.12.26, 0.12.25, 0.12.24, 0.12.23, 0.12.22)

root@host:/home/admin$ gem uninstall fluent-plugin-forest
Successfully uninstalled fluent-plugin-forest-0.3.0
root@host:/home/admin$ gem list fluentd*

*** LOCAL GEMS ***

fluent-mixin-config-placeholders (0.3.1)
fluent-plugin-loggly (0.0.7)
fluent-plugin-record-modifier (0.4.1)
fluentd (0.12.26, 0.12.25, 0.12.24, 0.12.23, 0.12.22)

root@host:/home/admin$ gem install fluent-plugin-forest
Fetching: fluent-plugin-forest-0.3.0.gem (100%)
ERROR:  Error installing fluent-plugin-forest:
	strptime requires Ruby version ~> 2.0.

```

TL;DR Fluentd is already there but installing a plugin without version restriction in the gemspec will result in the newest version of fluentd being wanted -> Broken system

File: fluent-plugin-sentry.gemspec (hunk 1 above)
Comment (ppp0) [inline fluent-plugin-sentry.gemspec]: I like this solution!

**Comment by y-ken** [comment]
It seems nice patch!

**Comment by ppp0** [comment]
Thank you, actually the build on travis is still failing for 1.9.3 - investigating

**Comment by y-ken** [comment]
Please sqush commit (git rebase -i HEAD~3)

**Comment by y-ken** [comment]
It is ok to create new PR and new branch.

**Comment by ppp0** [comment]
oh.. well, it might work now..? Sorry for the confusion!

**Comment by y-ken** [comment]
would you please make commit into one?

**Comment by ppp0** [comment]
sure, sorry

**Comment by y-ken** [comment]
Merged! thanks

**Comment by ppp0** [comment]
oh, you already released tag `v0.0.3`.. Hope this finds its home soon?

**Comment by y-ken** [comment]
It has released as 0.0.4!
https://rubygems.org/gems/fluent-plugin-sentry/versions/0.0.4

**Comment by y-ken** [comment]
I have fixed the issue with ruby 2.0 with following code.

``` ruby
  # Since Fluentd v0.14 requires ruby 2.1 or later.
  if defined?(RUBY_VERSION) && RUBY_VERSION < "2.1"
    spec.add_runtime_dependency "fluentd", "< 0.14"
  else
    spec.add_runtime_dependency "fluentd"
  end
```

**Comment by y-ken** [comment]
Hi @ppp0 
Do you have same problem to install `fluent-plugin-geoip`?
https://github.com/y-ken/fluent-plugin-geoip

**Comment by ppp0** [comment]
I did a test and yes, I'm afraid.. Same behaviour: `gem install` "fails" but plugin gets installed

``` sh
$ gem install fluent-plugin-geoip
Building native extensions.  This could take a while...
Fetching: fluent-plugin-geoip-0.6.1.gem (100%)
ERROR:  Error installing fluent-plugin-geoip:
    strptime requires Ruby version ~> 2.0.

$ gem list fluentd*

*** LOCAL GEMS ***

fluent-mixin-config-placeholders (0.3.1)
fluent-mixin-rewrite-tag-name (0.1.0)
fluent-plugin-forest (0.3.0)
fluent-plugin-geoip (0.6.1)
fluent-plugin-loggly (0.0.7)
fluent-plugin-record-modifier (0.4.1)
fluentd (0.12.26, 0.12.25, 0.12.24, 0.12.23, 0.12.22)
```

**Comment by ppp0** [comment]
Oh oh.. Things still don't work.. I tried

```
$gem install fluent-plugin-sentry --conservative
Fetching: fluent-plugin-sentry-0.0.5.gem (100%)
ERROR:  Error installing fluent-plugin-sentry:
    strptime requires Ruby version ~> 2.0.

$ gem list fluentd*

*** LOCAL GEMS ***

fluent-mixin-config-placeholders (0.3.1)
fluent-mixin-rewrite-tag-name (0.1.0)
fluent-plugin-forest (0.3.0)
fluent-plugin-loggly (0.0.7)
fluent-plugin-record-modifier (0.4.1)
fluent-plugin-sentry (0.0.5)
fluentd (0.12.26, 0.12.25, 0.12.24, 0.12.23, 0.12.22)
```

So, not even the `--conservative` option for `gem install` works 😿  - which according to the manual

```
--conservative               Don't attempt to upgrade gems already
                                     meeting version requirement
```

I'm running out of ideas - Thank you very much for your help!

**Comment by ppp0** [comment]
Moving away asap from Debian Wheezy

**Comment by y-ken** [comment]
strptime gem required ruby 2.0 or later. it won't work on ruby 1.9.
https://github.com/nurse/strptime

That gem has used since fluentd v0.14.x
It seems ruby 1.9 work only fluentd v0.10.x not fluentd v0.12.x or later.

**Comment by y-ken** [comment]
Did you resolved the issue that you had upgraded Debian Jessie or later from Wheezy?

**Comment by ppp0** [comment]
With Jessie, the aforementioned problems (strptime) are gone. But we have other problems that might have to do with our configuration's structure. We keep using fluentd 0.12.x - which works good on Wheezy except for that installation glitch.
